aioxmpp/service.py

```diff
@@ -347,45 +419,10 @@ class Fourth(metaclass=service.Meta):
     be instanciated before ``Foo``. There is no dependency relationship between
     ``Baz`` and ``Fourth``.
 
-    Inheritance works too::
-
-        class Foo(metaclass=service.Meta):
-            pass
-
-        class Bar(metaclass=service.Meta):
-            ORDER_BEFORE = [Foo]
-
-        class Baz(Bar):
-            # has ORDER_BEFORE == {Foo}
-            pass
-
-        class Fourth(Bar, inherit_dependencies=False):
-            # has empty ORDER_BEFORE
-            pass
-
     """
 
-    @classmethod
-    def transitive_collect(mcls, classes, attr, seen):
-        for cls in classes:
-            yield cls
-            yield from mcls.transitive_collect(getattr(cls, attr), attr, seen)
-
-    @classmethod
-    def collect_and_inherit(mcls, bases, namespace, attr,
-                            inherit_dependencies):
-        classes = set(namespace.get(attr, []))
-        if inherit_dependencies:
-            for base in bases:
-                if isinstance(base, mcls):
-                    classes.update(getattr(base, attr))
-        classes.update(
-            mcls.transitive_collect(
-                list(classes),
-                attr,
-                set())
-        )
-        return classes
+    dependency_graph = DependencyGraph()
+    service_order = {}
```

Comment:
I'll have to check, that it does not break the code. And mentioning it: The `_depgraph_node` member of the service classes should be documented (and perhaps uppercased, to fit with the other autogenerated memebers).